File: app/src/main/java/com/gnoemes/shikimoriapp/entity/app/domain/UserStatus.java
package com.gnoemes.shikimoriapp.entity.app.domain;

public enum UserStatus {
    AUTHORIZED("authorized"),
    GUEST("guest"),;

    private final String status;

    UserStatus(String status) {
        this.status = status;
    }

    public boolean isEqualStatus(String otherStatus) {
        return status.equals(otherStatus);
    }

    @Override
    public String toString() {
        return this.status;
    }
}

File: app/src/main/java/com/gnoemes/shikimoriapp/entity/app/domain/UserStatus.kt
package com.gnoemes.shikimoriapp.entity.app.domain

enum class UserStatus(val status: String) {
    AUTHORIZED("authorized"),
    GUEST("guest"), ;

    fun isEqualStatus(otherStatus: String): Boolean {
        return status == otherStatus
    }

    override fun toString(): String {
        return this.status
    }
}